### PR TITLE
Added simple lineinfile template

### PR DESCRIPTION
**Pull request by matejak**
This PR introduces a new template, and demonstrates its usage on fitting rules. Check out test scenarios for the `accounts_passwords_pam_faillock_enforce_local` rule (introduced in #6370) to see what configurations the template can handle.

This PR conflicts with #6303 and it should be merged after the template system is restructured. Then, it will need a rebase, so for now, consider this PR as a preview and forum about the template interface and capabilities.

### CODE REVIEW

**Review by jan-cerny — CHANGES_REQUESTED** [review]
Except for that it looks fine.

File: shared/templates/lineinfile/template.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+def lineinfile(data, lang):
```
[the remark this answers is not in the document]
Comment (matejak) [reply shared/templates/lineinfile/template.py]: This was really interesting, I have altered the template code to act more predictabley in cases like this.

**Pull request merged**